File: cpp_solutions/350.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        vector<int> res;
        vector<int> cnt(1001, 0);
        for (auto& num : nums1) {
            cnt[num] += 1;
        }

        for (auto& num : nums2) {
            if (cnt[num] > 0) {
                res.push_back(num);
                cnt[num] -= 1;
            }
        }
        return res;
    }
};

File: rust_solutions/350.rs
impl Solution {
    pub fn intersect(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<i32> {
        let mut res = Vec::new();
        let mut cnt = vec![0; 1001];

        for num in nums1 {
            cnt[num as usize] += 1;
        }
        for num in nums2 {
            if cnt[num as usize] > 0 {
                res.push(num);
                cnt[num as usize] -= 1;
            }
        }
        return res;
    }
}

